trio/_socket.py

```diff
@@ -472,44 +478,45 @@ def __getattr__(self, name):
             return getattr(self._sock, name)
         raise AttributeError(name)
 
-    def __dir__(self):
-        return super().__dir__() + list(self._forward)
+    def __dir__(self) -> Iterable[str]:
+        yield from super().__dir__()
+        yield from self._forward
```

Comment:
`dir()` isn't guaranteed to return a list - so seems bad to rely on that.

Reply:
https://docs.python.org/3/reference/datamodel.html#object.__dir__ says "a sequence must be returned", and the builtin `dir()` function converts it to a list.  Let's stick with the status quo here.

Reply:
Probably better to do `[*super().__dir__(), *self._forward]` then so we ourselves don't depend on it being a sequence.

Reply:
The status quo yields a typing error, since `super().__dir__()` is `Iterable[str]`. But sticking with returning a list to avoid unnecessary changes to the API is probably wise.